### PR TITLE
Add the dynamic ServiceAware functionality to injected Lists

### PR DESCRIPTION
**Pull request by stbischof**
from `osgi-test` mailing list

```
I was expecting that if I would have an injected List<S> with S a certain type of service, and I would register a service using the BundleContext in the test, the List<S> would reflect the newly registered service. A ServiceAware<S> also does not seem to reflect the new situation.

Are my expectations wrong?
```

This PR could solve this Problem but will break an other existing test.

@bjhargrave what do you think about?

### CODE REVIEW

File: org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import java.util.ListIterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+import org.osgi.test.common.service.ServiceAware;
+
+public class ServiceAwareList<E> implements List<E> {
```
Comment (bjhargrave) [inline org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java:16]: You could generalize this in to a `ListSupplierDelegate`.

```java
public class ListSupplierDelegate<E> implements List<E> {
    private final Supplier<? extends List<? extends E>> supplier;
    public ListSupplierDelegate(Supplier<? extends List<? extends E>> supplier) {
        this.supplier = Objects.requireNonNull(supplier);
    }
    @Override
	public void forEach(Consumer<? super E> action) {
		supplier.get()
			.forEach(action);
	}
```
 You create one with `new ListSupplierDelegate<>(configuration::getServices)`.

File: org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+	private ServiceAware<E> sa;
```
Comment (bjhargrave) [inline org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java:18]: This needs to be final

Comment (stbischof) [reply org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java]: ok

File: org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java (hunk 1 above)
Comment (stbischof) [reply org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java]: good idea

**Review by bjhargrave — APPROVED** [review]

**Comment by bjhargrave** [comment]
Nice! Thanks.

**Pull request merged**